commands/operator-sdk/cmd/scorecard/resource_handler.go

```diff
@@ -0,0 +1,253 @@
+			}
+		}
+		err = runtimeClient.Create(context.TODO(), obj)
+		if err != nil {
+			_, restErr := restMapper.RESTMappings(obj.GetObjectKind().GroupVersionKind().GroupKind())
+			if restErr == nil {
+				return err
+			}
+			// don't store error, as only error will be timeout. Error from runtime client will be easier for
+			// the user to understand than the timeout error, so just use that if we fail
+			wait.PollImmediate(time.Second*1, time.Second*10, func() (bool, error) {
+				restMapper.Reset()
+				_, err := restMapper.RESTMappings(obj.GetObjectKind().GroupVersionKind().GroupKind())
+				if err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			err = runtimeClient.Create(context.TODO(), obj)
+			if err != nil {
+				return err
+			}
+		}
+		addResourceCleanup(obj, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+	}
+	return nil
+}
+
+func createKubeconfigSecret() error {
+	kubeconfigMap := make(map[string][]byte)
+	kc, err := proxyConf.Create(metav1.OwnerReference{Name: "scorecard"}, "http://localhost:8888", SCConf.Namespace)
+	if err != nil {
+		return err
+	}
+	defer os.Remove(kc.Name())
+	kc, err = os.Open(kc.Name())
+	if err != nil {
+		return err
+	}
+	kcBytes, err := ioutil.ReadAll(kc)
+	if err != nil {
+		return err
+	}
+	kubeconfigMap["kubeconfig"] = kcBytes
+	kubeconfigSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "scorecard-kubeconfig",
+			Namespace: SCConf.Namespace,
+		},
+		Data: kubeconfigMap,
+	}
+	err = runtimeClient.Create(context.TODO(), kubeconfigSecret)
+	if err != nil {
+		return err
+	}
+	addResourceCleanup(kubeconfigSecret, types.NamespacedName{Namespace: kubeconfigSecret.GetNamespace(), Name: kubeconfigSecret.GetName()})
+	return nil
+}
+
+func addMountKubeconfigSecret(dep *appsv1.Deployment) {
+	// create mount for secret
+	dep.Spec.Template.Spec.Volumes = append(dep.Spec.Template.Spec.Volumes, v1.Volume{
+		Name: "scorecard-kubeconfig",
+		VolumeSource: v1.VolumeSource{Secret: &v1.SecretVolumeSource{
+			SecretName: "scorecard-kubeconfig",
+			Items: []v1.KeyToPath{{
+				Key:  "kubeconfig",
+				Path: "config",
+			}},
+		},
+		},
+	})
+	for index := range dep.Spec.Template.Spec.Containers {
+		// mount the volume
+		dep.Spec.Template.Spec.Containers[index].VolumeMounts = append(dep.Spec.Template.Spec.Containers[index].VolumeMounts, v1.VolumeMount{
+			Name:      "scorecard-kubeconfig",
+			MountPath: "/scorecard-secret",
+		})
+		// specify the path via KUBECONFIG env var
+		dep.Spec.Template.Spec.Containers[index].Env = append(dep.Spec.Template.Spec.Containers[index].Env, v1.EnvVar{
+			Name:  "KUBECONFIG",
+			Value: "/scorecard-secret/config",
+		})
+	}
+}
+
+func addProxyContainer(dep *appsv1.Deployment) {
+	dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers, v1.Container{
+		Name:            "scorecard-proxy",
+		Image:           "scorecard-proxy",
+		ImagePullPolicy: "Never",
```

Comment:
This has to be configurable I think. 

For many of the users, they will need the ability to set this, I think that we should do that now before merging this IMO

Reply:
Yeah. I had to change that locally for testing, and I guess it would make sense for that to be set to `Never` for CI or local testing as well. For end users, it should be set to `Always` by default. I'll make another flag for that as well.